### PR TITLE
Move to rest endpoint

### PR DESCRIPTION
**Pull request by yuvalbar**

### CODE REVIEW

**Comment by yuvalbar** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;T4CGiWw0&#x2F;61-pysdk-not-all-events-arrive"> PySDK - not all events arrive</a>

**Comment by yuvalbar** [comment]
Trigger Jenkins Build

File: alooma_pysdk/alooma_pysdk.py [hunk 1]
```diff
@@ -167,141 +143,131 @@ def __init__(self, token, servers='inputs.alooma.com', port=5001,
...
-
-        # Ensure JSON is newline-terminated
-        if event_wrapper[-1] != '\n':
-            event_wrapper += '\n'
+        # Wrap the event with metadata
+        event_wrapper[consts.WRAPPER_MESSAGE] = orig_event
 
         return event_wrapper
 
+    def report(self, event, metadata=None, block=None):
```
Comment (ofri) [inline alooma_pysdk/alooma_pysdk.py]: change to `block=False` to make it clearer that this is a boolean?

File: alooma_pysdk/alooma_pysdk.py [hunk 2]
```diff
@@ -167,141 +143,131 @@ def __init__(self, token, servers='inputs.alooma.com', port=5001,
...
+        """
+        Reports an event to Alooma by formatting it properly and placing it in
+        the buffer to be sent by the Sender instance
+        :param event:    A dict / string representing an event
+        :param metadata: (Optional) A dict with extra metadata to be attached to
+                         the event
+        :param block:    (Optional) Overrides the default SDK setting to block
```
Comment (ofri) [inline alooma_pysdk/alooma_pysdk.py]: The doc is not clear enough - what happens when block=True, what happens when block=False, and what is the default?

File: alooma_pysdk/alooma_pysdk.py [hunk 3]
```diff
@@ -167,141 +143,131 @@ def __init__(self, token, servers='inputs.alooma.com', port=5001,
...
+
+            return self._sender.enqueue_event(
+                    formatted_event, block if block else self.is_blocking)
+
+        else:  # Event is not a dict nor a string. Deny it.
+            error_message = (consts.LOG_MSG_BAD_EVENT % (type(event), event))
+            self._notify(consts.LOG_FAILED_SEND, error_message)
+            return False
+
+    def report_many(self, event_list, metadata=None, block=None):
```
Comment (ofri) [inline alooma_pysdk/alooma_pysdk.py]: block=False

File: alooma_pysdk/alooma_pysdk.py [hunk 4]
```diff
@@ -167,141 +143,131 @@ def __init__(self, token, servers='inputs.alooma.com', port=5001,
...
+        """
+        Reports all the given events to Alooma by formatting them properly and
+        placing them in the buffer to be sent by the Sender instance
+        :param event_list: A list of dicts / strings representing events
+        :param metadata: (Optional) A dict with extra metadata to be attached to
+                         the event
+        :param block:    (Optional) Overrides the default SDK setting to block
```
Comment (ofri) [inline alooma_pysdk/alooma_pysdk.py]: clearer docs

File: alooma_pysdk/alooma_pysdk.py [hunk 5]
```diff
@@ -382,125 +350,47 @@ def _choose_host(self):
                  connect
         """
         if len(self._hosts) == 1:
-            self._tcp_host = self._hosts[0]
+            self._http_host = self._hosts[0]
         else:
-            if self._tcp_host is None:
-                self._tcp_host = random.choice(self._hosts)
+            if self._http_host is not None:
```
Comment (ofri) [inline alooma_pysdk/alooma_pysdk.py]: why is the `not` here?

File: alooma_pysdk/alooma_pysdk.py [hunk 6]
```diff
@@ -382,125 +350,47 @@ def _choose_host(self):
...
-                else:
-                    self._notify(LOG_FAILED_SEND, e)
-                self._is_connected = False
-            except Exception, ex:  # Trap non-socket-error Exceptions.
-                self.enqueue_event(former_event)
-                self.enqueue_event(event)
-                self._notify(LOG_FAILED_SEND, ex)
-
-    def _enqueue_batches(self, batches):
+    def _enqueue_batches(self, *batches):
```
Comment (ofri) [inline alooma_pysdk/alooma_pysdk.py]: why do we need the `*`? Actually, why do we need a function that enqueues multiple batches? in the current implementation this is never used, so maybe get rid of it?

File: alooma_pysdk/alooma_pysdk.py [hunk 7]
```diff
@@ -96,48 +79,46 @@ class PythonSDK:
...
        function is called whenever a log-worthy event occurs, and is passed the
        event code and the proper message (see more details in the `callback`
        function documentation).
     """
 
-    def __init__(self, token, servers='inputs.alooma.com', port=5001,
-                 input_label='Python SDK', event_type=None, ssl_ca=_default_ca,
-                 callback=None, buffer_size=100000, blocking=True,
-                 batch_mode=True, batch_size=4096):
+    def __init__(self, token, servers=consts.DEFAULT_ALOOMA_ENDPOINT,
```
Comment (ofri) [inline alooma_pysdk/alooma_pysdk.py]: We should somehow document the `terminate` method and let the user know they must call it before their program terminates, or they risk losing the events in the last batch.

Comment (ofri) [inline alooma_pysdk/alooma_pysdk.py]: As yair pointed out - let's decide if we are OK with being backwards incompatible. If we are not, then let's make sure we handle unknown **kwargs.

File: alooma_pysdk/alooma_pysdk.py (hunk 1 above)
Comment (yuvalbar) [inline alooma_pysdk/alooma_pysdk.py]: done

File: alooma_pysdk/alooma_pysdk.py (hunk 2 above)
Comment (yuvalbar) [inline alooma_pysdk/alooma_pysdk.py]: Fixed the doc, kept the block parameter at `None` and fixed the bug in the `block` parameter check. Intended behaviour is - if `block` is `None`, use `self.is_blocking`, else use the value of `block`, be it `true` or `false`

File: alooma_pysdk/alooma_pysdk.py (hunk 3 above)
Comment (yuvalbar) [inline alooma_pysdk/alooma_pysdk.py]: See comment above

File: alooma_pysdk/alooma_pysdk.py (hunk 4 above)
Comment (yuvalbar) [inline alooma_pysdk/alooma_pysdk.py]: done

File: alooma_pysdk/alooma_pysdk.py (hunk 5 above)
Comment (yuvalbar) [inline alooma_pysdk/alooma_pysdk.py]: I made this code more readable

File: alooma_pysdk/alooma_pysdk.py (hunk 6 above)
Comment (yuvalbar) [inline alooma_pysdk/alooma_pysdk.py]: Cool. Made it `enqueue_batch`

File: alooma_pysdk/alooma_pysdk.py (hunk 1 above)
Comment (ofri) [inline alooma_pysdk/alooma_pysdk.py]: why not change it in the default value in the signature of the function as well?

**Pull request merged**